Fix Non-general conversions in acosh

aktech opened the pull request:
Currently `acosh()` have some non-general conversions, due to which it renders some wrong results.

The following is non-general:

``` python
In [10]: acosh(I*x)
Out[10]: I*acos(-x)

In [11]: acosh(-I*x)
Out[11]: I*acos(-x)

```

**Wrong Numerical Results**

``` python
In [5]: acosh(-5*I).n(5)
Out[5]: 2.2924 + 3.1416*I

In [6]: acosh(5*I).n(5)
Out[6]: 2.2924 + 3.1416*I

```

**References**:
1. WolframAlpha Results:
   - [`ArcCosh(-5*I)`](http://www.wolframalpha.com/input/?i=ArcCosh%28-5i%29)
   - [`ArcCosh(5*I)`](http://www.wolframalpha.com/input/?i=ArcCosh%285i%29)
   - [`I*ArcCos(-5)`](http://www.wolframalpha.com/input/?i=iArcCos%28-5%29)
2. StackOverflow: http://stackoverflow.com/questions/4788411/how-to-ask-mathematica-to-convert-arccosh-to-arccos
3. http://www.math10.com/en/algebra/hyperbolic-functions/hyperbolic-functions.html

@smichr  @asmeurer  Please have a look.

aktech committed:
Fix Non-general conversions in acosh

aktech commented:
@jksuom  
can you have a look at this?

jksuom commented:
@aktech 
From ``cosh(iy) = cos(y) = x`` if follows that ``acosh(x) = iy = i acos(x)``, so ``acosh(it) = i acos(it)``.
On the other hand, there is no simple relationship between ``acosh(it)`` and ``acos(t)``. So the code in master expressing the former in terms of the latter is wrong, and you are right in removing it.

However, I am not sure if adding the two imaginary infinities would be necessary.

aktech commented:
> However, I am not sure if adding the two imaginary infinities would be necessary.

That's not an extra addition, they are already present in tests (see tests [here](https://github.com/sympy/sympy/blob/master/sympy/functions/elementary/tests/test_hyperbolic.py#L479)). Since the lines I have removed, addressed these infinities earlier, that's why, I added them here as well, so as to avoid removing tests for them.

jksuom commented:
they are already present in tests
Yes, but perhaps it would be possible to find a more general condition to take care of all those cases,
something like ``arg.is_infinite``.

aktech commented:
Yes, that would be better, Fixed that.
Thanks!

jksuom commented:
Looks good now. +1
